axum/src/docs/routing/merge.md

```diff
@@ -37,7 +37,18 @@ let app = Router::new()
 # };
 ```
 
-## Panics
+# Merging routers with state
+
+When combining [`Router`]s with this function, each [`Router`] must have the
+same type of state. See ["Combining stateful routers"][combining-stateful-routers]
+for details.
+
+If you want to compose axum services with different types of state, use
+[`Router::nest_service`].
```

Comment:
That isn't possible. `merge` needs a `Router` because it literally copied the routes from one to the other and random services don't have routes.

https://github.com/tokio-rs/axum/pull/1552 is gonna make it work though.